### PR TITLE
docs: add Kuma to the list of implementations

### PR DESCRIPTION
**Pull request by lahabana**
/kind documentation

**What this PR does / why we need it**:

Add Kuma to the list of implementations of Gateway API

### CODE REVIEW

File: site-src/implementations.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ Implementors of Gateway API are encouraged to update this document with status i
 - [Kong][10] (alpha)
 - [NGINX Kubernetes Gateway][11] (pre-alpha)
 - [Traefik][12] (alpha)
+- [Kuma][13] (alpha)
```
Comment (hbagdi) [inline site-src/implementations.md:22]: This is an alphabetic list. Could you please put Kuma after Kong?

**Comment by youngnick** [comment]
Ping @lahabana, could you rebase and fix the sorting, then we can get this merged?

**Comment by lahabana** [comment]
@youngnick sorry I was on PTO. Updated and rebased

**Comment by robscott** [comment]
Thanks @lahabana!

/lgtm

**Pull request merged**